library/src/main/java/com/ricohapi/mstorage/MediaStorage.java

```diff
@@ -109,25 +110,41 @@ public void download(String mediaId, CompletionHandler<MediaContent> handler) {
 
     }
 
-//    public void list(Map<String, Objects> filter, CompletionHandler<MediaList> handler) {
-//        // TODO
-//    }
-
-    public void list(Map<String, String> params, CompletionHandler<MediaList> handler) {
+    public void list(Map<String, ?> params, CompletionHandler<MediaList> handler) {
         try {
             if (accessToken == null) {
                 throw new RicohAPIException(0, "wrong usage: use the connect method to get an access token.");
             }
             RicohAPIRequest request;
+            Map<String, String> header = new HashMap<>();
+            header.put("Authorization", "Bearer " + accessToken);
             if (params == null || params.isEmpty()) {
+                // GET /media
                 request = new RicohAPIRequest(ENDPOINT);
+                request.get(header);
             } else {
-                request = new RicohAPIRequest(ENDPOINT, params);
+                if (!params.containsKey(LIST_PARAM_KEY_FILTER) || params.get(LIST_PARAM_KEY_FILTER) == null) {
+                    // GET /media
+                    request = new RicohAPIRequest(ENDPOINT, params);
+                    request.get(header);
+                } else {
+                    // POST /media/search
+                    Map<String, Object> searchParams = new HashMap<>();
+                    searchParams.put("search_version", "2016-07-08");
+                    Map<String, Object> paging = new HashMap<>();
```

Comment:
I don't think so because of following reason:
* It will not need to modify this method when parameter of REST API will be changed such as more nested parameters will be available.
* If `param` is invalid format, the response from REST API will be error. So I think it's enough to return received response without checking `param`.